odpf/metadata/dataset.proto

```diff
@@ -1,5 +0,0 @@
-syntax = "proto3";
```

Comment:
@StewartJingga it seems like this was creating a problem while trying to generate go code from it. Since it is missing the `package` specifier and the `go_package` option. Is this being used somewhere as of now ?

Reply:
@rohilsurana This file can be completely removed.